torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -451,6 +451,232 @@ class IrParser {
+            }
+
+            TensorView* running_var = nullptr;
+            if (!node->input(4)->type()->isSubtypeOf(
+                    static_cast<c10::TypePtr>(NoneType::get()))) {
+              running_var =
+                  value_map[node->input(4)->unique()]->as<TensorView>();
+            }
+
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            auto training = constant_as<bool>(node->input(5));
+            TORCH_INTERNAL_ASSERT(
+                training.has_value(),
+                "The training (bool) parameter is required.");
+            const bool kTraining = training.value();
+
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            auto momentum = constant_as<float>(node->input(6));
+            TORCH_INTERNAL_ASSERT(
+                momentum.has_value(),
+                "The momentum (float) parameter is required.");
+            const float kMomentum = momentum.value();
+
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers)
+            auto eps = constant_as<float>(node->input(7));
+            TORCH_INTERNAL_ASSERT(
+                eps.has_value(), "The EPS parameter is required.");
+            const float kEps = eps.value();
+
+            // TODO: NAN when mean and variance are zero
+            // --ftz=true -- flush-to-zero
+
+            const int kNumberOfDims = input->nDims();
+            std::vector<int> reduction_axes;
+            std::vector<bool> broadcast_mask(kNumberOfDims, false);
+            Val* num_features = nullptr;
+            for (size_t axis = 0; axis < kNumberOfDims; ++axis) {
+              if (axis != 1) {
+                reduction_axes.push_back(axis);
+                broadcast_mask[axis] = true;
+                num_features = (num_features == nullptr)
+                    ? input->domain()->domain()[0]->extent()
+                    : mul(num_features,
+                          input->domain()->domain()[axis]->extent());
+              }
+            }
+
+            // Algorithm
+            auto x_sum = sum(input, reduction_axes);
+            auto x_sum_bcast = broadcast(x_sum, broadcast_mask);
+            auto x_mean = div(x_sum_bcast, num_features);
+
+            /*
+            auto current_mean_hat = mul(x_mean, new Float(kMomentum));
+            auto rmean_bcast = broadcast(running_mean, broadcast_mask);
+            auto mean_hat = mul(rmean_bcast, new Float(1.0 - kMomentum));
+            auto new_mean_hat = add(mean_hat, current_mean_hat);
+            */
+
+            auto x_mean_sub = sub(input, x_mean);
+            auto x_mean_sub_pow =
+                binaryOp(BinaryOpType::Pow, x_mean_sub, new Float(2));
+            auto var_sum = sum(x_mean_sub_pow, reduction_axes);
+            auto var_sum_bcast = broadcast(var_sum, broadcast_mask);
+            auto var = div(var_sum_bcast, num_features);
+
+            /*
+            auto current_var_hat = mul(var, new Float(kMomentum));
```

Comment:
Somehow my comments are not posted.......

I think this is intended for running_var update? Should use unbiased variance for running stats.
```
auto unbiased_var = div(var_sum_bcast, sub(num_features, new Int(1)))
auto current_var_hat = mul(unbiased_var, new Float(kMomentum));
```

Reply:
:+1: